service/controller/rbac/resource/namespaceauth/create.go

```diff
@@ -144,8 +144,8 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 	return nil
 }
 
-func needsUpdate(role role, existingRoleBinding *rbacv1.RoleBinding) bool {
-	return role.targetGroup != existingRoleBinding.Subjects[0].Name
+func needsUpdate(desiredRoleBinding, existingRoleBinding *rbacv1.RoleBinding) bool {
+	return desiredRoleBinding.Subjects[0].Name != existingRoleBinding.Subjects[0].Name
```

Comment:
Is the subject the only thing that can change?

Reply:
Also: Is it guaranteed that there's always at least one element in `Subjects`?

Reply:
I changed the code so that it takes the subject namespace and the roleref into consideration, as those are the things that we care about.